erpnext-fhir-mapper/src/main/java/com/ozonehis/eip/mappers/erpnext/QuotationItemMapper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.mappers.erpnext;
+
+import com.ozonehis.eip.mappers.ToERPNextMapping;
+import com.ozonehis.eip.model.erpnext.QuotationItem;
+import org.hl7.fhir.r4.model.Dosage;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QuotationItemMapper<R extends Resource> implements ToERPNextMapping<R, QuotationItem> {
+
+    @Override
+    public QuotationItem toERPNext(R resource) {
+        QuotationItem quotationItem = new QuotationItem();
+        if (resource instanceof ServiceRequest serviceRequest) {
+            quotationItem.setCustomExternalID(serviceRequest.getIdPart());
+            if (serviceRequest.hasCode()) {
+                quotationItem.setItemCode(
+                        serviceRequest.getCode().getCodingFirstRep().getCode());
+            }
+            quotationItem.setQuantity(1); // default quantity is 1 for serviceRequests.
+            String requesterDisplay = serviceRequest.getRequester().getDisplay();
+            String serviceDisplay = serviceRequest.getCode().getText();
+            quotationItem.setDescription(serviceDisplay + " | Requester: " + requesterDisplay);
+
+        } else if (resource instanceof MedicationRequest medicationRequest) {
+            quotationItem.setCustomExternalID(medicationRequest.getIdPart());
+            Quantity quantity = medicationRequest.getDispenseRequest().getQuantity();
+            quotationItem.setQuantity(quantity.getValue().intValue());
+            quotationItem.setUnitOfMeasure(quantity.getUnit());
+            String medicationCode =
+                    medicationRequest.getMedicationReference().getReference().split("/")[1];
+            quotationItem.setItemCode(medicationCode);
+            String requesterDisplay = medicationRequest.getRequester().getDisplay();
+            String medicationDisplay =
+                    medicationRequest.getMedicationReference().getDisplay();
+            quotationItem.setDescription(medicationDisplay + " | Requester: " + requesterDisplay);
+            // Add dosage instructions to the notes.
+            quotationItem.setNotes(constructDosageInstructionsText(medicationRequest));
+
+        } else {
```

Comment:
OMRS orders do have a comment to the fulfiller, which gets mapped to the `MedicationRequest.note`. That might need to be preserved here?

Reply:
Let's put a pin on that. Need to customize ERPNext to add more custom fields.